space-cli/main.go

```diff
@@ -1,115 +1,112 @@
 package main
 
 import (
+	"fmt"
 	"os"
 
-	"github.com/sirupsen/logrus"
-	"github.com/urfave/cli"
-
-	"github.com/spaceuptech/space-cli/modules"
 	"github.com/spaceuptech/space-cli/modules/addons"
-	"github.com/spaceuptech/space-cli/modules/auth"
-	"github.com/spaceuptech/space-cli/modules/database"
-	"github.com/spaceuptech/space-cli/modules/deploy"
-	"github.com/spaceuptech/space-cli/modules/eventing"
-	"github.com/spaceuptech/space-cli/modules/filestore"
-	"github.com/spaceuptech/space-cli/modules/ingress"
-	"github.com/spaceuptech/space-cli/modules/letsencrypt"
-	"github.com/spaceuptech/space-cli/modules/operations"
-	"github.com/spaceuptech/space-cli/modules/project"
-	remoteservices "github.com/spaceuptech/space-cli/modules/remote-services"
-	"github.com/spaceuptech/space-cli/modules/services"
-	"github.com/spaceuptech/space-cli/modules/userman"
-	"github.com/spaceuptech/space-cli/utils"
+	"github.com/spf13/cobra"
 )
 
+var rootCmd = &cobra.Command{
+	Use:     "space-cli",
+	Version: "0.16.0",
+}
+
+func execute() error {
+	return rootCmd.Execute()
+}
+
+func init() {
+	//cobra.OnInitialize(initConfig)
+	rootCmd.PersistentFlags().StringP("log-level", "", "info", "Sets the log level of the command")
+	addons.AddCmd.PersistentFlags().StringP("project", "", "", "The project to add the add-on to")
+	addons.RemoveCmd.PersistentFlags().StringP("project", "", "", "The project to remove the add-on from")
```

Comment:
You should set this in `addons/commands.go`. This file will contain a `Commands` function which will return an array of commands.

Reply:
Also, project needs to be a root level persistent flag

Reply:
Use viper as well.